svc/mediaservice/transport.go

```diff
@@ -0,0 +1,136 @@
+package mediaservice
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/SatorNetwork/sator-api/internal/utils"
+	"github.com/thedevsaddam/govalidator"
+
+	"github.com/SatorNetwork/sator-api/internal/httpencoder"
+
+	"github.com/go-chi/chi"
+	jwtkit "github.com/go-kit/kit/auth/jwt"
+	"github.com/go-kit/kit/transport"
+	httptransport "github.com/go-kit/kit/transport/http"
+)
+
+// Predefined request query keys
+const (
+	pageParam         = "page"
+	itemsPerPageParam = "items_per_page"
+)
+
+type (
+	logger interface {
+		Log(keyvals ...interface{}) error
+	}
+)
+
+// MakeHTTPHandler ...
+func MakeHTTPHandler(e Endpoints, log logger) http.Handler {
+	r := chi.NewRouter()
+
+	options := []httptransport.ServerOption{
+		httptransport.ServerErrorHandler(transport.NewLogErrorHandler(log)),
+		httptransport.ServerErrorEncoder(httpencoder.EncodeError(log, codeAndMessageFrom)),
+		httptransport.ServerBefore(jwtkit.HTTPToContext()),
+	}
+
+	r.Post("/", httptransport.NewServer(
+		e.AddItem,
+		decodeAddItemRequest,
+		httpencoder.EncodeResponse,
+		options...,
+	).ServeHTTP)
+
+	r.Get("/", httptransport.NewServer(
+		e.GetItemsList,
+		decodeGetItemsListRequest,
+		httpencoder.EncodeResponse,
+		options...,
+	).ServeHTTP)
+
+	r.Get("/{id}", httptransport.NewServer(
+		e.GetItemByID,
+		decodeGetItemByIDRequest,
+		httpencoder.EncodeResponse,
+		options...,
+	).ServeHTTP)
+
+	r.Delete("/{id}", httptransport.NewServer(
+		e.DeleteItemByID,
+		decodeDeleteItemByIDRequest,
+		httpencoder.EncodeResponse,
+		options...,
+	).ServeHTTP)
+
+	return r
+}
+
+func decodeGetItemsListRequest(_ context.Context, r *http.Request) (interface{}, error) {
+	return PaginationRequest{
+		Page:         castStrToInt32(r.URL.Query().Get(pageParam)),
+		ItemsPerPage: castStrToInt32(r.URL.Query().Get(itemsPerPageParam)),
+	}, nil
+}
+
+func castStrToInt32(source string) int32 {
+	res, err := strconv.Atoi(source)
+	if err != nil {
```

Comment:
@dmitrymomot  do I need here sth like this?

func castStrToInt32(source string) int32 {
	parsed, err := strconv.ParseInt(source, 10, 32)
	if err != nil {
		return 0
	}
	
	return int32(parsed)
}